sources/packages/backend/apps/api/src/route-controllers/restriction/_tests_/e2e/restriction.institution.controller.getStudentRestrictions.e2e-spec.ts

```diff
@@ -0,0 +1,239 @@
+    // Arrange
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    const student = await saveFakeStudent(appDataSource);
+    const application = createFakeApplication({
+      location: collegeCLocation,
+      student,
+    });
+    await applicationRepo.save(application);
+    await saveFakeStudentRestriction(
+      appDataSource,
+      { student, application, creator: student.user },
+      {
+        restrictionCategory: "Verification",
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    const endpoint = `/institutions/restriction/student/${student.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN);
+  });
+
+  it("Should return forbidden 403 error code when a BC Public Institution User tries to get student restrictions for a student which they don't have access to.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(appDataSource);
+    await saveFakeStudentRestriction(
+      appDataSource,
+      { student, creator: student.user },
+      {
+        restrictionCategory: "Verification",
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/restriction/student/${student.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN);
+  });
+
+  it("Should not get the student restriction when the notification type has a value of `No effect` and restriction category has a value of `Verification`.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(appDataSource);
+    const application = createFakeApplication({
+      location: collegeFLocation,
+      student,
+    });
+    await applicationRepo.save(application);
+    await saveFakeStudentRestriction(
+      appDataSource,
+      { student, application, creator: student.user },
+      {
+        restrictionCategory: "Verification",
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/restriction/student/${student.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
```

Comment:
Two restrictions can be created for the student, one with `No effect` and other one to match the criteria to be returned.

Reply:
Is this change implemented?

Reply:
Yes, for the 2nd half, I am finding a restriction (any restriction) from the db that has `notificationType` as `RestrictionNotificationType.Warning`. Once retrieved, I am creating a `fakeStudentRestriction` and then hitting the endpoint to make sure that it can be retrieved. For the 1st half, I am finding a restriction (any restriction) from the db that has `notificationType` as `RestrictionNotificationType.NoEffect`. Once retrieved, I am creating a `fakeStudentRestriction` and then hitting the endpoint to make sure that it cannot be retrieved.